### PR TITLE
Edit of the code. 

### PR DESCRIPTION
**Pull request by ToastedXi**
Firstly, change from Display name usage to modelname usage. The reason for this is that modelnames are unique and therefore allow more configuration and avoid situations where multiple vehicles might have the same display name and cause problems with the config.
It is also a very simple change to migrate from with any modern text editing tool and should require less than 5 minutes to switch from

Thanks!

### CODE REVIEW

**Comment by arkhunter623** [comment]
I know there are people currently using the system in it's current state to set tones department wide without having an insanely long CFG(good example would be fire in a server I dev. Has over 80 cars)

**Comment by ToastedXi** [comment]
> I know there are people currently using the system in it's current state to set tones department wide without having an insanely long CFG(good example would be fire in a server I dev. Has over 80 cars)

I understand your response, However the point i am trying to push it not to have shorter CFGs its to have a more unique aspect of this script.

**Comment by arkhunter623** [comment]
Correct. However uniqueness can still be obtained with game names. I understand where your coming from with wanting to use model names but for some servers changing is hours of work.

**Comment by arkhunter623** [comment]
Maybe a configuration option to select witch one your using is possible? To allow for both at the end users choice

**Comment by xfunnypigx** [comment]
I firmly believe that it behooves everyone to have a sustainable and configurable way of handling their servers.
Changing all models to have unique displaynames when unique modelnames already exist is _not_ sustainable.
Rather, changing the config once and having an easy way of extending it from there, is. Especially new users would benefit from this.

That being said, @arkhunter623 's suggestion to have a configuration option would be a decent idea. Unsure if OP is willing to expand on it themselves, though.

**Comment by ToastedXi** [comment]
> Correct. However uniqueness can still be obtained with game names. I understand where your coming from with wanting to use model names but for some servers changing is hours of work.

If this was taken into consideration it wouldn't change the old code. For servers that are using it already, It would be an updated version so if the old version works then you would still be able to use that perfectly fine.

**Comment by arkhunter623** [comment]
A couple of things I was seeing. Is this based of the latest master or a fork? Also Not sure if you deleted the comment or it's not showing for me @ToastedXi so I pulled a copy of it from the email GitHub sent.
"If this was taken into consideration it wouldn't change the old code. For servers that are using it already, It would be an updated version so if the old version works then you would still be able to use that perfectly fine." 
Basically saying here take this update that for some servers will require large amounts of work to change or not get new features.... Right.... I'm not going to comment further on this. This is up to Trevor wether to implement or not.

**Comment by ToastedXi** [comment]
> A couple of things I was seeing. Is this based of the latest master or a older fork? Also Not sure if you deleted the comment or it's not showing for me @ToastedXi so I pulled a copy of it from the email GitHub sent.
> "If this was taken into consideration it wouldn't change the old code. For servers that are using it already, It would be an updated version so if the old version works then you would still be able to use that perfectly fine."
> Basically saying here take this update that for some servers will require large amounts of work to change or not get new features.... Right.... I'm not going to comment further on this. This is up to Trevor wether to implement or not.

I deleted the following post i put due to being incorrect.

**Comment by TrevorBarns** [comment]
I'm sorry I'm reviewing on mobile rn, is there any actual code changes here? Looks to be version comparison between master and stable. What native do you recommend to get Model Name?

**Comment by arkhunter623** [comment]
> I'm sorry I'm reviewing on mobile rn, is there any actual code changes here? Looks to be version comparison between master and stable. What native do you recommend to get Model Name? 

I believe that this was done off a old fork. I looked through the code and did see some changes but this is nowhere near up to date with latest master. Thus probably why it's giving the unable to cleanly merge

**Comment by xfunnypigx** [comment]
There are actual changes here that would differentiate it from current master, yeah. Namely changing from 

`veh_name = GetDisplayNameFromVehicleModel(GetEntityModel(veh))` to
`veh_name = GetEntityModel(veh)`

This would result in using the vehicle model name, rather than the display name, which can be prone to overlaps.
Arkhunter suggested that maybe this could be a toggle so servers would have less trouble migrating.
But as he also pointed out, there would definitely be conflicts. Given how small the changes are, though (namely 2, one pointed out above and the other being changing the `'<name>'` to ` instead (couldn't for the life of me figure out how to format it there, sorry), it should be a relatively easy fix if this is decided to be added.

**Comment by TrevorBarns** [comment]
I understand you are a first time contributor, thank you for providing input on LVC. ❤️ Please note the differences between pull requests and issues though. Issues is an area for bugs, suggestions, improvements, help request, etc. Pull request on the other hand is where contributors are requesting to merge improvements to edited code. This discussion belongs under 'issues'.
> It is also a very simple change to migrate from with any modern text editing tool and should require less than 5 minutes to switch from

For example, if it is so easy to implement, a pull request would effectively prove your point by offering the lines that need to be edited and allows for very rapid testing and potentially deployment. 

I highly recommend reevaluating the tone used when working on GitHub. No need to delete comments or add any emotion to arguments. This is a place of collaboration and teamwork, thanks! 

---

That being said, using `GetEntityModel(veh)` has some drawbacks. It returns the hash of the vehicle in hex/decimal not string. For example, `GetEntityModel(veh)` where veh is an custom car with model name 'so3' would return `-1248899570`. 

In order for that to work efficiently, we would need to iterate through all tables and convert their string keys to hash keys.

```
SIREN_ASSIGNMENTS = {
       ['so3'] = { },      --after--> ['-1248899570'] = { },
       ['pd2'] = { },      --after--> ['-1839648114'] = { },
}
```

This isn't so bad as it could be done on script start up. But, a feature introduced in 3.2.3-beta vehicle model wildcarding wouldn't work. Using hashes you cannot simply create wild cards like 'so#' to assign sirens to vehicles like 'so1','so2',...'so99'. 

The second part to this is, why not use display names? I mean, sure, it is a little overhead for the first installation of LVC. But, developers already have to manually set model names, so any new car added to the server after the first setup is as simple as copying the model name to the display/game name. The overhead for changing all display names can be fixed within 30 minutes using a smart notepad like Notepad++. Less if you have another developer help you. 

Potential For Choice:
@arkhunter623 brought up a good point that maybe there could be the choice of using either. This adds some support overhead: determining whether a user is in the correct mode before we can even start. But, would not be too hard to implement and is something I will consider for a future update. The only real argument I could see for model names is if a server uses accurate/custom display names in other scripts.

TLDR:
- Please be nice and remain as professional as possible. 
- `GetEntityModel(veh)` returns hash not string, removing ability for wildcards, and easier configuration for large fleets.
- Display name conflicts, to me, are not huge issue, and can easily be resolved. 

---
Please reply with your thoughts as I am interested to hear your feedback. 

Regards,

**Comment by xfunnypigx** [comment]
> That being said, using `GetEntityModel(veh)` has some drawbacks. It returns the hash of the vehicle in hex/decimal not string. For example, `GetEntityModel(veh)` where veh is an custom car with model name 'so3' would return `-1248899570`.
> 
> In order for that to work efficiently, we would need to iterate through all tables and convert their string keys to hash keys.
> 
> ```
> SIREN_ASSIGNMENTS = {
>        ['so3'] = { },      --after--> ['-1248899570'] = { },
>        ['pd2'] = { },      --after--> ['-1839648114'] = { },
> }
> ```
> 
> This isn't so bad as it could be done on script start up. But, a feature introduced in 3.2.3-beta vehicle model wildcarding wouldn't work. Using hashes you cannot simply create wild cards like 'so#' to assign sirens to vehicles like 'so1','so2',...'so99'.

Thanks for the insight on this, I wasn't actually aware it was this involved.

> The second part to this is, why not use display names? I mean, sure, it is a little overhead for the first installation of LVC. But, developers already have to manually set model names, so any new car added to the server after the first setup is as simple as copying the model name to the display/game name. The overhead for changing all display names can be fixed within 30 minutes using a smart notepad like Notepad++. Less if you have another developer help you.
> 
> Potential For Choice:
> @arkhunter623 brought up a good point that maybe there could be the choice of using either. This adds some support overhead: determining whether a user is in the correct mode before we can even start. But, would not be too hard to implement and is something I will consider for a future update. The only real argument I could see for model names is if a server uses accurate/custom display names in other scripts.


As for the latter, here is an example

Let us suppose we are adding a new police cruiser. From a creator standpoint, giving it the modelname <mycoolcruiser> and giving it a display name of <POLICE> is not unheard of. Using these pre-existing hashes is easier on the creator and there isn't a real reason *not* to use them. Unless a user is interested in it being one specific one of the million crown vic models, it is both serviceable as well as more streamlined to use. In other words, why fix it, if it ain't broke.

Model names by necessity are unique. On one hand, using 10 different Police Cruisers with the display name <POLICE> can make setup easier, however, it can also lead to situations where.. uhh.. let us say we have 2 different Police Bikes for different departments with the same display name, leading to an impasse.

Now one could argue that they could just change the display name and that is fair to an extent, but the way I think of it is a bit like saying "Let us change the badge from  Ford Explorer to Ford Explorer 155311. Possible, but perhaps not necessarily.. smart, for a lack of better terms.
Meanwhile, working with the VIN would be more useful in that case.

To put it short: It would be easier and smarter to do config changes, than for creators and server owners alike to use by definition non-unique identifiers, especially when these unique identifiers already exist and require no extra changes on the outside, only on a config level. Instead of changing values in two places, you are only changing them in one, which goes back to my comment on sustainability.

Of course, though, there are situations where using the display name can be useful. Using wildcards, as well as "abusing" the fact that vehicles may not be unique can be good. Using <POLICE#> effectively adds many different vehicles at once, which can make configs shorter. And even when not using wildcards, going back to my example, but assuming they are *not* from different departments, simply adding POLICEB would add them both.

In the end, I suppose, it's about user choice and giving the user tools to work. If they want a simplistic approach and potentially a shorter approach, they can use that and it makes sense for them.
If they want the fine-tuning ability without necessarily having to change the vehicles themselves, that would then be possible, as well.

And to be clear, I'm not dragging on the mod or anything, both of these approaches have their uses, their advantages, and disadvantages.
